modules/internal/ChapelArray.chpl

```diff
@@ -1571,16 +1569,18 @@ module ChapelArray {
       if super.type != this.type then
         compilerError("isSubset called with different associative domain types");
 
+      if isRectangularDom(this) then
+        return && reduce forall i in 1..this.dims().size do
```

Comment:
Since dims().size is expected to be quite small, I think it makes more sense to write this out as a for loop, rather than a reduction over a forall